indico/modules/events/papers/controllers/management.py

```diff
@@ -111,3 +113,78 @@ def _process(self):
+    """Assign contrubutions to reviewers and judges"""
+
+    def _process(self):
+        self.list_generator = PaperAssignmentListGenerator(event=self.event_new)
+        return WPManagePapers.render_template('management/assignment.html', self._conf, event=self.event_new,
+                                              **self.list_generator.get_list_kwargs())
+
+
+class RHAssignmentListCustomize(RHManagePapersBase):
+    """Filter options and columns to display for the paper assignment list of an event"""
+
+    def _checkParams(self, params):
+        RHManagePapersBase._checkParams(self, params)
+        self.list_generator = PaperAssignmentListGenerator(event=self.event_new)
+
+    def _process_GET(self):
+        list_config = self.list_generator.list_config
+        return WPManagePapers.render_template('management/assignment_list_filter.html', self._conf,
+                                              event=self.event_new,
+                                              static_items=self.list_generator.static_items,
+                                              filters=list_config['filters'],
+                                              visible_items=list_config['items'])
+
+    def _process_POST(self):
+        self.list_generator.store_configuration()
+        return jsonify_data(flash=False, **self.list_generator.render_list())
+
+
+class RHManagePapersActionsBase(RHManagePapersBase):
+    """Base class for RHs performing actions on selected paper contributions"""
+
+    _contrib_query_options = ()
+
+    @property
+    def _contrib_query(self):
+        query = Contribution.query.with_parent(self.event_new)
+        if self._contrib_query_options:
+            query = query.options(*self._contrib_query_options)
+        return query
+
+    def _checkParams(self, params):
+        RHManagePapersBase._checkParams(self, params)
+        self.list_generator = PaperAssignmentListGenerator(event=self.event_new)
+        ids = map(int, request.form.getlist('contribution_id'))
+        self.contributions = self._contrib_query.filter(Contribution.id.in_(ids)).all()
+
+
+class RHBulkPaperStateChange(RHManagePapersActionsBase):
+    """Perform bulk change state operations on selected papers"""
+
+    def _process(self):
+        form = BulkPaperJudgmentForm(event=self.event_new, judgment=request.form.get('judgment'),
+                                     contribution_id=[a.id for a in self.contributions])
+        if form.validate_on_submit():
+            judgment_data, contrib_data = form.split_data
+            # TODO: For now, I'm considering that any submitted paper contribution can change
+            # its state (even rejected ones)
+            submitted_contributions = [c for c in self.contributions if c.paper_last_revision]
+            for contrib in submitted_contributions:
+                paper_change_state(contrib, contrib_data, judge=session.user, **judgment_data)
+            num_judged_contribs = len(submitted_contributions)
```

Comment:
It's the papers that are judged, not the contributions. I would name these `num_judged_papers`.